src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -956,119 +957,150 @@ private static void ExpandTypeName(TypeSyntax type, StringBuilder builder)
             }
         }
 
-        private static string GetContainerDisplayName(SyntaxNode node)
+        private string GetContainerDisplayName(SyntaxNode node)
         {
-            return GetContainer(node, immediate: true);
+            return GetDisplayName(node, DisplayNameOptions.IncludeTypeParameters);
         }
 
-        private static string GetFullyQualifiedContainerName(SyntaxNode node)
+        private string GetFullyQualifiedContainerName(SyntaxNode node)
         {
-            return GetContainer(node, immediate: false);
+            return GetDisplayName(node, DisplayNameOptions.IncludeNamespaces);
         }
 
-        private static string GetContainer(SyntaxNode node, bool immediate)
+        public string GetDisplayName(SyntaxNode node, DisplayNameOptions options, string rootNamespace = null)
         {
             if (node == null)
             {
                 return string.Empty;
             }
 
-            var name = GetNodeName(node, includeTypeParameters: immediate);
-            var names = new List<string> { name };
+            var pooled = PooledStringBuilder.GetInstance();
```

Comment:
Typically, we don't (unless the try/finally is intended to allow us to recover from some "expected" Exception).  If the code below throws, we're going to fail fast, so in this case, there's really no point.